pkg/manifests/manifests.go

```diff
@@ -1430,6 +1430,10 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 		p.Spec.Retention = f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.Retention
 	}
 
+	if f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize != "" && f.config.ClusterMonitoringConfiguration.PrometheusK8sConfig.RetentionSize != "0" {
```

Comment:
Going back to this I'm not sure that we need to treat "0" in a special way.

Reply:
I removed the check for 0